TileTowns/Assets/Scripts/Managers/RoadManager.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Tilemaps;
+using Zenject;
+
+public class RoadManager : MonoBehaviour
+{
+    private event Action<ITileData> RoadPlaced;
+
+    [Inject] private IGameManager _gameManager;
+    [Inject] private IBuildingController _buildingController;
+    [Inject] protected ITileMapData TileMapData { get; private set; }
+
+    [SerializeField] public Tile upDownLeftRightRoad;
+    [SerializeField] public Tile upLeftRightRoad;
+    [SerializeField] public Tile downLeftRightRoad;
+    [SerializeField] public Tile upDownLeftRoad;
+    [SerializeField] public Tile upDownRightRoad;
+    [SerializeField] public Tile upRightRoad;
+    [SerializeField] public Tile downLeftRoad;
+    [SerializeField] public Tile upDownRoad;
+    [SerializeField] public Tile downRightRoad;
+    [SerializeField] public Tile upLeftRoad;
+    [SerializeField] public Tile leftRightRoad;
+    [SerializeField] public Tile upRoad;
+    [SerializeField] public Tile rightRoad;
+    [SerializeField] public Tile downRoad;
+    [SerializeField] public Tile leftRoad;
+    
+    private Tilemap _tilemap;
+    private Dictionary<Vector2Int, bool> _roadMap = new();
+    private const int RoadZIndex = 1;
+    
+    private void Start()
+    {
+        _gameManager.LevelLoaded += InitializeRoadMap;
+        _gameManager.LevelLoaded += OnGameLoaded;
+        
+        RoadPlaced += CreateRoad;
+        _buildingController.PlacedBuilding += (_, tileData) => RoadPlaced?.Invoke(tileData);
```

Comment:
you sure its not better to just make a method OnBuildingPlaced()?
this doesnt seem too readable

Reply:
Events are for communicating between objects, not inside them ;)